core/shared/src/main/scala/zio/prelude/NonEmptyMap.scala

```diff
@@ -0,0 +1,224 @@
+package zio.prelude
+
+/*
+ * Copyright 2020-2022 John A. De Goes and the ZIO Contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import zio.NonEmptyChunk
+
+import scala.language.implicitConversions
+
+/**
+ * A non-empty wrapper for the scala immutable map. Note - this does not attempt to implement all features of
+ * map but what the author considers to be the "normal ones".
+ */
+case class NonEmptyMap[K, V] private (private val map: Map[K, V]) { self =>
```

Comment:
```suggestion
case class NonEmptyMap[K, V] private (private val map: Map[K, V]) extends AnyVal { self =>
```
any reason not to?

Reply:
sorry I don't understand what you want there or why

Reply:
@regiskuckaertz is suggesting a change to make this into a value class via `extends AnyVal`
![image](https://user-images.githubusercontent.com/9019485/170696282-a35fb364-7538-4378-87bf-7e631f36447a.png)

Reply:
That being said, I don't think we want to do that. For example because of `equals`, we want that to be true only when both sides are `NonEmptyMap`

Reply:
but why?